### PR TITLE
Fix Potential PHP Notices/Warnings in JInstallerHelper class

### PR DESCRIPTION
**Pull request by bassmanpaul**
@see https://github.com/joomla/joomla-cms/issues/6000

I experienced a scenario where I was testing an install package (which was corrupt) and it brought to light a couple of PHP notices in the `JInstallerHelper` class.
#### The File:

**libraries/cms/installer/helper.php** line 197 & 221
#### Issue 1 (line 197):

``` php
$dirList = array_merge(JFolder::files($extractdir, ''), JFolder::folders($extractdir, ''));
```

As `JFolder::files()` & `JFolder::folders()` can return `false` which would break `array_merge()`. One solution would be type casting:

``` php
$dirList = array_merge((array)JFolder::files($extractdir, ''), (array)JFolder::folders($extractdir, ''));
```
#### Issue 2 (line 221):

``` php
if (!count($files))
```

Again `JFolder::files()` could return `false` so the `count()` of `false` would be **1** as opposed to **0**. (see: http://php.net/manual/en/function.count.php#example-5167) Therefore as this is a false positive may I suggest something like:

``` php
if (!$files || !count($files))
```

or

``` php
if (!is_array($files) || !count($files))
```

If everyone likes the suggestions I will happily create a Pull Request.

### CODE REVIEW

File: libraries/cms/installer/helper.php [hunk 1]
```diff
@@ -172,7 +172,7 @@ public static function unpack($p_filename, $alwaysReturnArray = false)
 		 * List all the items in the installation directory.  If there is only one, and
 		 * it is a folder, then we will set that folder to be the installation folder.
 		 */
-		$dirList = array_merge(JFolder::files($extractdir, ''), JFolder::folders($extractdir, ''));
+		$dirList = array_merge((array)JFolder::files($extractdir, ''), (array)JFolder::folders($extractdir, ''));
```
Comment (zero-24) [inline libraries/cms/installer/helper.php]: @bassmanpaul can you add a space between `(array)` and `JFolder::` to make travis happy?

File: libraries/cms/installer/helper.php [hunk 2]
```diff
@@ -218,7 +218,7 @@ public static function detectType($p_dir)
 		// Search the install dir for an XML file
 		$files = JFolder::files($p_dir, '\.xml$', 1, true);
 
-		if (!count($files))
+		if(!$files || !count($files))
```
Comment (zero-24) [inline libraries/cms/installer/helper.php]: @bassmanpaul can you readd the space after the if? So we have

`if (!$files || !count($files))`

Than travis should be happy with us :smile: Thanks

**Comment by zero-24** [comment]
@bassmanpaul i have just add your description from your report and add comments about codestyle to fix travis. I hope you can have a look into it.

Regarding Travis see: https://travis-ci.org/joomla/joomla-cms/jobs/50025400

**Comment by wilsonge** [comment]
Merged on review. Thankyou very much for the fix :)

**Pull request merged**